### PR TITLE
add missing execution space in mdrange

### PR DESCRIPTION
**Pull request by yasahi-hpc**
This PR aims at fixing bugs about execution space in MDRange in some internal helpers.
For example, following modifications are made

```C++
// Previous, exec_space was missing
range_type range(
    point_type{{0, 0}}, point_type{{len0, len1}}, tile_type{{4, 4}}
    // [TO DO] Choose optimal tile sizes for each device
 );

// After fix
range_type range(
    exec_space, point_type{{0, 0}}, point_type{{len0, len1}},
    tile_type{{4, 4}}  // [TO DO] Choose optimal tile sizes for each device
);
```

### CODE REVIEW

File: common/src/KokkosFFT_Helpers.hpp [hunk 1]
```diff
@@ -179,7 +179,7 @@ namespace KokkosFFT {
 ///
 /// \return Sampling frequency
 template <typename ExecutionSpace, typename RealType>
-auto fftfreq(const ExecutionSpace& exec_space, const std::size_t n,
+auto fftfreq(const ExecutionSpace&, const std::size_t n,
```
Comment (cedricchevalier19) [inline common/src/KokkosFFT_Helpers.hpp:182]: Perhaps it is better to leave this for another PR.

File: common/src/KokkosFFT_Helpers.hpp [hunk 2]
```diff
@@ -214,7 +214,7 @@ auto fftfreq(const ExecutionSpace& exec_space, const std::size_t n,
 ///
 /// \return Sampling frequency starting from zero
 template <typename ExecutionSpace, typename RealType>
-auto rfftfreq(const ExecutionSpace& exec_space, const std::size_t n,
+auto rfftfreq(const ExecutionSpace&, const std::size_t n,
```
Comment (cedricchevalier19) [inline common/src/KokkosFFT_Helpers.hpp:217]: Perhaps it is better to leave this for another PR.

File: common/src/KokkosFFT_transpose.hpp [hunk 3]
```diff
@@ -113,8 +113,8 @@ void _prep_transpose_view(InViewType& in, OutViewType& out,
 
 template <typename ExecutionSpace, typename InViewType, typename OutViewType,
           std::enable_if_t<InViewType::rank() == 1, std::nullptr_t> = nullptr>
-void _transpose(const ExecutionSpace& exec_space, InViewType& in,
-                OutViewType& out, [[maybe_unused]] axis_type<1> _map) {}
+void _transpose(const ExecutionSpace&, InViewType& in, OutViewType& out,
+                [[maybe_unused]] axis_type<1> _map) {}
```
Comment (cedricchevalier19) [inline common/src/KokkosFFT_transpose.hpp:117]: Perhaps it is better to leave this for another PR.

**Review by cedricchevalier19 — APPROVED** [review]
Good catch.

Arg name deletion can be done as another PR.

**Comment by yasahi-hpc** [comment]
> Good catch.
> 
> Arg name deletion can be done as another PR.

Thank you for your review. I have modified accordingly.
Arg name deletion may be done in your PR?
